lib/widgets/dialog.dart

```diff
@@ -61,28 +77,31 @@ DialogStatus showErrorDialog({
   return DialogStatus(future);
 }
 
-void showSuggestedActionDialog({
+/// Displays an alert dialog with a cancel button and an action button.
+///
+/// The [DialogStatus.closed] Future gives true if the action button was tapped.
+/// If the dialog was canceled,
+/// either with the cancel button or by tapping outside the dialog's area,
+/// it completes with null.
+DialogStatus<bool> showSuggestedActionDialog({
   required BuildContext context,
   required String title,
   required String message,
   required String? actionButtonText,
-  required VoidCallback onActionButtonPress,
 }) {
   final zulipLocalizations = ZulipLocalizations.of(context);
-  showDialog<void>(
+  final future = showDialog<bool>(
     context: context,
     builder: (BuildContext context) => AlertDialog(
       title: Text(title),
       content: SingleChildScrollView(child: Text(message)),
       actions: [
         TextButton(
-          onPressed: () => Navigator.pop(context),
+          onPressed: () => Navigator.pop<bool?>(context, null),
```

Comment:
nit: the types allow `bool` here (and below):
```suggestion
          onPressed: () => Navigator.pop<bool>(context, null),
```

and that seems a bit more consistent with `bool` as the type parameter on `showDialog`